.github/workflows/reusable-python-publish_package.yml

```diff
@@ -0,0 +1,171 @@
+            required: true
+            type: string
+
+    secrets:
+        EXTRA_PYPI_REPO_USER:
+            description: 'Extra PyPI repository username for dependencies. Only used if `extra_repository` is True.'
+            required: false
+        EXTRA_PYPI_REPO_TOKEN:
+            description: 'Extra PyPI repository token. Only used if `extra_repository` is True.'
+            required: false
+        MIRA_REPO_SECRETS:
+            description: 'Base64 encoded Yanml file content, containing all the repository secrets'
+            required: true
+
+defaults:
+    run:
+        shell: 'bash -l {0}'
+
+jobs:
+    release_pypi_package:
+        name: Release PyPI package
+        runs-on: ubuntu-latest
+        timeout-minutes: 20
+        if: ${{ inputs.package_manager != 'conda' }}
+        env:
+            cache_number: 1
+        steps:
+            -   uses: actions/checkout@v4
+                with:
+                    lfs: ${{ inputs.lfs }}
+                    fetch-depth: 0
+            -   name: Set up Python version
+                uses: actions/setup-python@v5
+                with:
+                    python-version: ${{ inputs.python_version }}
+            -   uses: MiraGeoscience/CI-tools/.github/actions/reusable-python-setup_poetry@main
+                name: Setup poetry env
+                if: ${{ inputs.package_manager == 'poetry' }}
+                with:
+                    python_ver: ${{ inputs.python_version }}
+                    cache_number: ${{ env.cache_number}}
+                    runner_os: ${{ runner.os }}
+                    extra_repository: ${{ inputs.extra_repository }}
+                    repo_user: ${{ secrets.EXTRA_PYPI_REPO_USER }}
+                    repo_token: ${{ secrets.EXTRA_PYPI_REPO_TOKEN }}
+            -   uses: MiraGeoscience/CI-tools/.github/actions/reusable-python-setup_hatch@main
+                name: Setup hatch env
+                if: ${{ inputs.package_manager == 'hatch' }}
+                with:
+                    cache_number: ${{ env.cache_number}}
+                    runner_os: ${{ runner.os }}
+            -   name: Build package
+                run: |
+                    if ${{ inputs.package_manager == 'poetry' }}; then
+                        poetry build
+                    elif ${{ inputs.package_manager == 'hatch'}}; then
+                        hatch build
+                    else
+                        echo "Unknown pypi package manager. ${{ inputs.package_manager }} is not part of ['hatch', 'poetry']."
+                        exit 1
+                    fi
+            -   name: Publish
+                run: |
+                    MIRA_REPO_CONFIG=$(echo '${{ secrets.MIRA_REPO_SECRETS }}' | base64 -d)
+                    repository_list=$(echo '${{ inputs.repo_names }}' | jq -r '.[]')
+                    for repo_name in ${repository_list}; do
```

Comment:
if this step was turned into a re-usable action, could it define a matrix from the repository_list ?

Reply:
anyway, good as is for now. Maybe just apply changes for paranoïd mode 😉  using `$(echo ${val})`